tests/frame/concat_test.py

```diff
@@ -10,10 +10,10 @@ def test_concat_horizontal(constructor: Any, request: Any) -> None:
     if "dask" in str(constructor):
         request.applymarker(pytest.mark.xfail)
     data = {"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.0, 8, 9]}
-    df_left = nw.from_native(constructor(data))
+    df_left = nw.from_native(constructor(data)).lazy()
 
     data_right = {"c": [6, 12, -1], "d": [0, -4, 2]}
-    df_right = nw.from_native(constructor(data_right))
+    df_right = nw.from_native(constructor(data_right)).lazy()
```

Comment:
Solution was to cast both to lazy so that mypy knows that it's not either eager or lazy.

Reply:
ah I remember noticing this some time ago, then forgot about it...